c7n/resources/ebs.py

```diff
@@ -50,8 +50,51 @@ class SnapshotAge(AgeFilter):
         days={'type': 'number'},
         op={'type': 'string', 'enum': OPERATORS.keys()})
     date_attribute = 'StartTime'
-
-
+    
+
+def _filter_ami_snapshots(self, snapshots):
+    c = local_session(self.manager.session_factory).client('ec2')
+    amis = self.manager._cache.get(
+        {'region': self.manager.config.region,
+         'resource': 'ami'})
+    # try using cache first to get a listing of all AMI snapshots and compares resources to the list
+    if amis is None:
+        # This will populate the cache.
+        ami_manager = AMI(self.manager.ctx, {})
+        amis = ami_manager.resources()
+        ami_snaps = []
+        data = c.describe_images(Owners=['self'])['Images']
```

Comment:
you can basically replace all of this ami lookup including the cache get with 

        ami_manager = AMI(self.manager.ctx, {})
        amis = ami_manager.resources()